src/openrct2-ui/windows/ServerList.cpp

```diff
@@ -73,529 +71,519 @@ static Widget window_server_list_widgets[] = {
-{
-    int32_t port = NETWORK_DEFAULT_PORT;
-    auto beginBracketIndex = address.find('[');
-    auto endBracketIndex = address.find(']');
-    auto dotIndex = address.find('.');
-    auto colonIndex = address.find_last_of(':');
-    if (colonIndex != std::string::npos)
-    {
-        if (endBracketIndex != std::string::npos || dotIndex != std::string::npos)
-        {
-            auto ret = std::sscanf(&address[colonIndex + 1], "%d", &port);
-            assert(ret);
-            if (ret > 0)
+            // Draw compatibility icon
+            right -= 10;
+            int32_t compatibilitySpriteId;
+            if (serverDetails.Version.empty())
             {
-                address = address.substr(0, colonIndex);
+                // Server not online...
+                compatibilitySpriteId = SPR_G2_RCT1_CLOSE_BUTTON_0;
             }
-        }
-    }
+            else
+            {
+                // Server online... check version
+                bool correctVersion = serverDetails.Version == NetworkGetVersion();
+                compatibilitySpriteId = correctVersion ? SPR_G2_RCT1_OPEN_BUTTON_2 : SPR_G2_RCT1_CLOSE_BUTTON_2;
+            }
+            GfxDrawSprite(&dpi, ImageId(compatibilitySpriteId), { right, screenCoords.y + 1 });
+            right -= 4;
 
-    if (beginBracketIndex != std::string::npos && endBracketIndex != std::string::npos)
-    {
-        address = address.substr(beginBracketIndex + 1, endBracketIndex - beginBracketIndex - 1);
-    }
+            // Draw lock icon
+            right -= 8;
+            if (serverDetails.RequiresPassword)
+            {
+                GfxDrawSprite(&dpi, ImageId(SPR_G2_LOCKED), { right, screenCoords.y + 4 });
+            }
+            right -= 6;
 
-    if (!NetworkBeginClient(address, port))
-    {
-        ContextShowError(STR_UNABLE_TO_CONNECT_TO_SERVER, STR_NONE, {});
-    }
-}
+            // Draw number of players
+            screenCoords.x = right - numPlayersStringWidth;
+            GfxDrawString(dpi, screenCoords + ScreenCoordsXY{ 0, 3 }, players, { colours[1] });
 
-static void ServerListFetchServersBegin()
-{
-    if (_fetchFuture.valid())
-    {
-        // A fetch is already in progress
-        return;
+            screenCoords.y += ITEM_HEIGHT;
+        }
     }
 
-    _serverList.Clear();
-    _serverList.ReadAndAddFavourites();
-    _statusText = STR_SERVER_LIST_CONNECTING;
-
-    _fetchFuture = std::async(std::launch::async, [] {
-        // Spin off background fetches
-        auto lanF = _serverList.FetchLocalServerListAsync();
-        auto wanF = _serverList.FetchOnlineServerListAsync();
+#    pragma endregion
 
-        // Merge or deal with errors
-        std::vector<ServerListEntry> allEntries;
-        try
-        {
-            auto entries = lanF.get();
-            allEntries.insert(allEntries.end(), entries.begin(), entries.end());
-        }
-        catch (...)
+private:
+    void JoinServer(std::string address)
```

Comment:
Sounds like this doesn't use any data of the class itself, and could just be a free function on this source file.

Reply:
Yeah that's correct, makes sense - will move it outside of the class.